modules/sops/default.nix

```diff
@@ -79,6 +79,13 @@ let
           User of the file.
         '';
       };
+      uid = lib.mkOption {
```

Comment:
Looks good but can you add an assertion to the assertions nixos options that make sure we are not setting both uid and owner?

Reply:
Overall I think we should prefer `uid` if it's not-null and `owner` != `root`. Otherwise it's a bit clunky if no `root` is set. The other alternative would be to switch to `uid` == 0 by default and make `owner` nullable.

Reply:
Also let me think about this case a bit.

Reply:
I changed the code to follow your second option. I think this easier to implement.